src/urllib3/response.py

```diff
@@ -1020,10 +1020,18 @@ def stream(
             yield from self.read_chunked(amt, decode_content=decode_content)
         else:
             while not is_fp_closed(self._fp) or len(self._decoded_buffer) > 0:
-                data = self.read(amt=amt, decode_content=decode_content)
-
+                if amt is None:
+                    data = self.read1(amt=amt, decode_content=decode_content)
+                else:
+                    data = self.read(amt=amt, decode_content=decode_content)
                 if data:
                     yield data
+                else:
+                    if self._fp:
+                        self._fp.close()
```

Comment:
Please add a comment explaining in which situation `self._fp` is not closed by `http.client`

Reply:
I don't have a good feeling for when it isn't closed, but have commented on the case you pointed out

Reply:
This comment makes me very apprehensive about merging this and also makes me worry about the correctness of the `read1` PR.